syft/pkg/cataloger/golang/parse_go_bin.go

```diff
@@ -0,0 +1,125 @@
+package golang
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/anchore/syft/syft/pkg"
+)
+
+const packageIdentifier = "dep"
+
+func parseGoBin(path string, reader io.ReadCloser) ([]pkg.Package, error) {
+	pkgsSlice := make([]pkg.Package, 0)
+
+	// Identify if bin was compiled by go
+	x, err := openExe(reader)
+	if err != nil {
+		return pkgsSlice, err
+	}
+
+	_, mod := findVers(x)
+	fields := strings.Fields(mod)
+
+	// slice off root pacakge info
+	var separator int
+	for x, field := range fields {
+		if field == packageIdentifier {
```

Comment:
open question: do we want to only preserve dependencies? Or should we keep a package for the root module (without a version since it isn't available)?

Reply:
Let's stick with preserving dependencies for the time being. I don't think a document would be accurate if the software sourced itself as a dependency

Reply:
I am adding a `Location` to each package so that there is additional metadata on the path where this was found. Not always analogous to the root package, but let me know if you have a good spot in mind to place the root package.